src/parameterized_tendencies/microphysics/microphysics_wrappers.jl

```diff
@@ -117,105 +126,96 @@ Also returns the total energy source term due to the microphysics processes.
 function compute_precipitation_sources!(
     Sᵖ,
     Sᵖ_snow,
-    Sqₜᵖ,
+    Sqₗᵖ,
+    Sqᵢᵖ,
     Sqᵣᵖ,
     Sqₛᵖ,
-    Seₜᵖ,
     ρ,
     qᵣ,
     qₛ,
     ts,
-    Φ,
     dt,
     mp,
     thp,
 )
     FT = eltype(thp)
     # @. Sqₜᵖ = FT(0) should work after fixing
     # https://github.com/CliMA/ClimaCore.jl/issues/1786
-    @. Sqₜᵖ = ρ * FT(0)
+    @. Sqₗᵖ = ρ * FT(0)
+    @. Sqᵢᵖ = ρ * FT(0)
     @. Sqᵣᵖ = ρ * FT(0)
     @. Sqₛᵖ = ρ * FT(0)
-    @. Seₜᵖ = ρ * FT(0)
+
+    @. qᵣ = max(FT(0), qᵣ)
+    @. qₛ = max(FT(0), qₛ)
```

Comment:
I think you will change the input to this function (e.g. c_specific.q_rai) if you do this - is this what you want?

Reply:
Good point. I'll re-write it so that I don't overwrite the `specific` fields